tools/dataset_converters/yolo2coco.py

```diff
@@ -0,0 +1,162 @@
+
+    yolo_label_dir = osp.join(image_dir, 'labels')
+    yolo_image_dir = osp.join(image_dir, 'images')
+    yolo_class_txt = osp.join(image_dir, 'classes.txt')
+    with open(yolo_class_txt) as f:
+        classes = f.read().strip().split()
+
+    indices = os.listdir(yolo_image_dir)
+
+    if split:
+        print(
+            'Start to organize data according to existing train/test/val split'
+        )
+        train_dataset = {'images': [], 'annotations': [], 'categories': []}
+        val_dataset = {'images': [], 'annotations': [], 'categories': []}
+        test_dataset = {'images': [], 'annotations': [], 'categories': []}
+
+        # categories id starts from 0
+        for i, cls in enumerate(classes, 0):
+            train_dataset['categories'].append({'id': i, 'name': cls})
+            val_dataset['categories'].append({'id': i, 'name': cls})
+            test_dataset['categories'].append({'id': i, 'name': cls})
+        train_img, val_img, test_img = format_by_file(image_dir)
+    else:
+        print('Start to organize data')
+        dataset = {'images': [], 'annotations': [], 'categories': []}
+        for i, cls in enumerate(classes, 0):
+            dataset['categories'].append({'id': i, 'name': cls})
+
+    obj_count = 0
+    for idx, file in enumerate(mmengine.track_iter_progress(indices)):
+        # support both .jpg and .png
+        txtFile = file.replace('images',
+                               'txt').replace('.jpg',
+                                              '.txt').replace('.png', '.txt')
+        img = mmcv.imread(osp.join(image_dir, 'images/') + file)
+        height, width = img.shape[:2]
+
+        if split:
+            if file in train_img:
+                dataset = train_dataset
+            elif file in val_img:
+                dataset = val_dataset
+            elif file in test_img:
+                dataset = test_dataset
+
+        dataset['images'].append({
+            'file_name': file,
+            'id': idx,
+            'width': width,
+            'height': height
+        })
+
+        if not osp.exists(osp.join(yolo_label_dir, txtFile)):
+            # if current image is not annotated
+            continue
+
+        with open(osp.join(yolo_label_dir, txtFile)) as f:
+            labels = f.readlines()
+            for label in labels:
+                label = label.strip().split()
+                x = float(label[1])
+                y = float(label[2])
+                w = float(label[3])
+                h = float(label[4])
+
+                # convert x,y,w,h to x1,y1,x2,y2
+                H, W = img.shape[:2]
+                x1 = (x - w / 2) * W
+                y1 = (y - h / 2) * H
+                x2 = (x + w / 2) * W
+                y2 = (y + h / 2) * H
+
+                cls_id = int(label[0])
+                width = max(0, x2 - x1)
+                height = max(0, y2 - y1)
+                dataset['annotations'].append({
+                    'image_id':
```

Comment:
-> `'image_id`: idx,

Reply:
Seems this part is modified by the lint check during the commit, I will find a way to walk through it.

Reply:
> -> `'image_id': idx,`

Seems this part is modified by the lint check during the commit, I will find a way to walk through it.